app/src/main/java/org/mozilla/focus/browser/BrowserFragment.kt

```diff
@@ -127,8 +127,9 @@ class BrowserFragment : IWebViewLifecycleFragment() {
                         }
                         NavigationEvent.VAL_UNCHECKED -> {
                             url.toUri()?.let {
-                                if (BundledTilesManager.getInstance(context).unpinSite(context, it)) {
-                                    browserOverlay.removePinnedSiteFromTiles(it.toString())
+                                val bundledTileId = BundledTilesManager.getInstance(context).unpinSite(context, it)
+                                if (bundledTileId != null) {
+                                    browserOverlay.removePinnedSiteFromTiles(bundledTileId)
```

Comment:
This has some code smells here - it already more than doubles the previous code length, makes duplicate calls, and checks a bunch of cases. It also causes code complexity down the line (by making it handle two types of inputs) and also passes around `null` as an expected value that needs to be checked. (Using `null` to signal state is not a good idea). [You should read that Clean Code book :P ]

Can you think of a cleaner way to structure this code?

I'd encourage you not to "just write code that works" or "fix problems created in previous commits" - I think it's easy to do that, but if code looks messy, consider how you can clean it up.